### ISSUE TITLE
Two easy questions.

### ISSUE THREAD

**Issue by DarkAlchy** [bmaltais/kohya_ss#1017]
First question is: what do we lose when extracting a LyCORIS from a DB model?
Second question is: when are we getting a must-have for 2.x feature (that the DB method already has) of stopping TE training at xx%?

**Comment by bmaltais** [comment]
1. Good question. Some a making their LyCORIS from DB extraction and they work quite well... so I think as long as the method you use give you satisfactory result it should be fine.
2. Are you talking about the lack of support for stop TE training at %? I will check. Last I did it was not yet supported in kohya_ss sd-script code... but he might have introduced support for it recently.

**Comment by DarkAlchy** [comment]
Well, the reason I asked about #1 is interrelated to #2.

#1 is because I cannot get a L based training to work no matter what I have tried while a DB training with the same data works BUT when I extract it the resulting Lycoris, or Lora, is far far weaker.  Sometimes my DB training is weak but works then I extract and it becomes so weak I have to :2, or :3 weight it to see it.  Which leads into #2 because I can't get any L to train in 2.x, but I did with DB due to being able to stop that TE (falls between 25-50% for me).

**Comment by bmaltais** [comment]
Try extracting a Kohya LoCon at 256 rank and alpha (essentially 256 everywhere) is it better? Then try to resize it with fro 0.9 strength and see if it still work. It should give you a reasonably sized Lora.

**Comment by DarkAlchy** [comment]
Locon or Lycoris?  I will try that but even 64/1, 64/32 was weak.  How do I resize it with this 0.9?

Oh, did you notice that in DB the loss rates are so much less than with a L anything?

**Comment by bmaltais** [comment]
Try the Kohya LoCon at 256/256, 256,256

Your LoRa are be wise you use alpha 1. Keep the alpha the same as the rank.

**Comment by DarkAlchy** [comment]
Going to give you an example of the model vs a 1.2GB 512/512

![image](https://github.com/bmaltais/kohya_ss/assets/10561690/12810b8a-0609-4651-83b5-92b9048f0703)
![image](https://github.com/bmaltais/kohya_ss/assets/10561690/be13f72c-0285-4fa9-bb86-dafb86cf3b12)

next is your 256/256 and the one after a 384/384

![image](https://github.com/bmaltais/kohya_ss/assets/10561690/02b403fc-1534-4d4a-8b7d-6dfd853b02c9)
![image](https://github.com/bmaltais/kohya_ss/assets/10561690/f8971961-0b5b-4d5d-a6d0-c0f34d9a557c)

**Comment by DarkAlchy** [comment]
![image](https://github.com/bmaltais/kohya_ss/assets/10561690/042eb2d5-42e1-4fa9-92c9-459a31067e38)
I do not see a Kohya Locon only the LyCORIS version.

**Comment by bmaltais** [comment]
So I guess the top is what the DB model is producing... and the 3 belod (all looking similar) is the LoRA... YEah... quite a difference indeed...

As far as the Kohya_ss LoCon it is this option it is the "Extract LoRA" Tab... My bad, I am too used to the LoRA trainer GUI.

**Comment by DarkAlchy** [comment]
Is this extract Lora or is this Extract Kohya Locon since both are different?

![image](https://github.com/bmaltais/kohya_ss/assets/10561690/792bd1dc-e8b9-43cf-a7eb-9ba368cfc4b1)

**Comment by bmaltais** [comment]
For the lack of stop TE ofter % you could try setting the coption dropout to a high value... it should provide a good approximation... for example, iy you are stopping the TE training after 25%... set the caption dropout to 0.75... this will essentially train the caption only 25% of the total training... Might be worth a shot... and who knows... might even produce better results than DB.

**Comment by bmaltais** [comment]
> Is this extract Lora or is this Extract Kohya Locon since both are different?
> 
> ![image](https://user-images.githubusercontent.com/10561690/246953036-792bd1dc-e8b9-43cf-a7eb-9ba368cfc4b1.png)

This is what I meant by extrach Kohya LoCon... this extract method essentially produce a LoCon as an output.

**Comment by DarkAlchy** [comment]
Then there is no true extract LoRA it is one step higher and is really a Locon?

I have to admit I never played with that drop out stuff before.

**Comment by bmaltais** [comment]
The caption dropout will essentially drop caption training for the specified ratio... to sort of kinda like stopping training after x% of steps... except it is done during the whole training... so might technically be better.

**Comment by DarkAlchy** [comment]
I'll give it a shot and there is something different between extract lora and lycoris because I ran out of cuda ram with it while lycoris I did up to 512/512.  I switched to CPU.

**Comment by DarkAlchy** [comment]
Here is something interesting but 384/384 Lycoris is identical (as identical as xformers will allow) as 256/256 Locon.  Size wise it was larger.

LyCORIS then Locon
![image](https://github.com/bmaltais/kohya_ss/assets/10561690/c11d4cd9-cf78-4a76-9d24-470049d67f62)
![image](https://github.com/bmaltais/kohya_ss/assets/10561690/5e5e4575-fa0a-4529-a179-35e89d8c918c)

**Comment by DarkAlchy** [comment]
I trained with the drop out to 50%, which is what I used for my DB above.

![image](https://github.com/bmaltais/kohya_ss/assets/10561690/cfb0d1fe-c8b3-479a-ab75-0b468e4958f1)

**Comment by bmaltais** [comment]
Interesting... Getting closer... Try the Prodigy optimizer with LR set to 1 everywhere...

**Comment by DarkAlchy** [comment]
Well, I tried this on another prompt and it tore it up.  shaking my head on these L things as DB just seems to work it is when I extract I lose too much.

**Comment by DarkAlchy** [comment]
Prodigy and 1 everywhere. 
![image](https://github.com/bmaltais/kohya_ss/assets/10561690/903ee899-dcba-4d85-8fa2-c210d62221e3)

**Comment by bmaltais** [comment]
Not good

**Comment by DarkAlchy** [comment]
Yeah, I didn't like it as it took all my vram so I was forced to turn on checkpointing just to be able to get BS1 on a 16GB T4.  I can say it was not worth it though the author does say it is was meant for cosine with annealing that I asked Kohya to implement many months ago.  Without that it isn't as good the author was saying.  A few other don't dos as well (like warmup for the first steps).

**Comment by bmaltais** [comment]
Yeah, Prodigy is a VRAM eater for sure. I can hardly batch of 4 640,640 on a 3090.

**Comment by DarkAlchy** [comment]
I just wonder if it is worth it?  We really will not know until Kohya adds the cosine with annealing that this was made for.  As it stands it just isn't worth it.

**Comment by bmaltais** [comment]
Well... it did give some good results in some of my models... but you need to compensate the small train batch size with the gradient accumulation... So say I used to do batch of 8 with AdamW and can only do 4 with Prodigy... then I will set the gradient accumulate 2 to get the same effect as batch 8.

**Comment by DarkAlchy** [comment]
I use BS1, or 2 depending, and AdamW so what would I do then?  The fact I have to turn checkpointing on at all is the murderer as 20m becomes 32m.  While locally that wouldn't matter so much on Colab/Runpod that time differences is precious (especially on Colab).

**Comment by DarkAlchy** [comment]
I was waiting on an answer from you to close this, but thanks.
